src/Simplex/Chat.hs

```diff
@@ -162,19 +162,25 @@ newChatController ChatDatabase {chatStore, agentStore} user cfg@ChatConfig {agen
   showLiveItems <- newTVarIO False
   pure ChatController {activeTo, firstTime, currentUser, smpAgent, agentAsync, chatStore, chatStoreChanged, idsDrg, inputQ, outputQ, notifyQ, chatLock, sndFiles, rcvFiles, currentCalls, config, sendNotification, incognitoMode, filesFolder, expireCIsAsync, expireCIs, cleanupManagerAsync, timedItemThreads, showLiveItems}
   where
-    configServers :: InitialAgentServers
+    configServers :: DefaultAgentServers
     configServers =
-      let smp' = fromMaybe (smp defaultServers) (nonEmpty smpServers)
+      let smp' = fromMaybe (smp (defaultServers :: DefaultAgentServers)) (nonEmpty smpServers)
        in defaultServers {smp = smp', netCfg = networkConfig}
     agentServers :: ChatConfig -> IO InitialAgentServers
-    agentServers config@ChatConfig {defaultServers = ss@InitialAgentServers {smp}} = do
-      smp' <- maybe (pure smp) userServers user
-      pure ss {smp = smp'}
+    agentServers config@ChatConfig {defaultServers = DefaultAgentServers {smp, ntf, netCfg}} = do
+      users <- withTransaction chatStore getUsers
+      smp' <- case users of
+        [] -> pure $ M.fromList [(1, smp)]
+        _ -> serversMap users
+      pure InitialAgentServers {smp = smp', ntf, netCfg}
       where
+        serversMap :: [User] -> IO (M.Map UserId (NonEmpty SMPServerWithAuth))
+        serversMap users = M.fromList <$> mapM (\user' -> (aUserId user',) <$> userServers user') users
```

Comment:
```suggestion
        initialServers :: [User] -> IO [UserId, NonEmpty SMPServerWithAuth]
        initialServers = mapM (\u -> (aUserId u,) <$> userServers u)
```

Reply:
Int64 is here because it is not a chat UserId, we could of course use imported/namespaced UserId from Agent but it's overkill.